src/kafka/connection/connection_manager.hh

```diff
@@ -0,0 +1,115 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Copyright (C) 2019 ScyllaDB Ltd.
+ */
+
+#pragma once
+
+#include "kafka_connection.hh"
+#include "../protocol/metadata_response.hh"
+#include "../protocol/metadata_request.hh"
+
+#include <boost/functional/hash.hpp>
+
+#include <unordered_map>
+
+namespace seastar {
+
+namespace kafka {
+
+struct metadata_refresh_exception : public std::runtime_error {
+public:
+    metadata_refresh_exception(const std::string& message) : runtime_error(message) {}
+};
+
+class connection_manager {
+public:
+
+    using connection_id = std::pair<std::string, uint16_t>;
```

Comment:
It would be more kosher to create an object instead of packing hostname and port into a pair.

Reply:
We only ever access the fields inside to pass them down to `kafka_connection::connect()` where we pass both of them anyway. Apart from that, we just use `{host, port}` as lookup keys. Is it really worth it to create a separate struct for it?